napari/_qt/qt_modal.py

```diff
@@ -0,0 +1,53 @@
+from qtpy.QtCore import Qt, QPoint
+from qtpy.QtWidgets import (
+    QVBoxLayout,
+    QPushButton,
+    QDialog,
+    QFormLayout,
+    QFrame,
+)
+from qtpy.QtGui import QCursor
+
+
+class QtModalPopup(QDialog):
+    """A generic modal popup window.
+
+    The seemingly extra frame here is to allow rounded corners on a truly
+    transparent background
+
+    +-------------------------------
+    | Dialog
+    |  +----------------------------
+    |  | QVBoxLayout
+    |  |  +-------------------------
+    |  |  | QFrame
+    |  |  |  +----------------------
+    |  |  |  | QFormLayout
+    |  |  |  |
+    |  |  |  |  (stuff goes here)
+    """
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.setObjectName("QtModalPopup")
+        self.setModal(True)
+        self.setWindowFlags(Qt.Popup | Qt.FramelessWindowHint)
+        self.setLayout(QVBoxLayout())
+
+        self.frame = QFrame()
+        self.frame.setObjectName("QtPopupFrame")
+        self.layout().addWidget(self.frame)
+        self.layout().setContentsMargins(0, 0, 0, 0)
+
+        self.form_layout = QFormLayout()
+        closebutton = QPushButton("Close")
```

Comment:
How do you feel about dropping the explicit close button, but then if you click anywhere outside the pop-up it closes, sort of like right clicking in chrome or other applications which bring up those dropdown like menus.

Reply:
I like that idea... I actually tried briefly to add it.  failed on the first attempt and then never got back to it.  I can take another look though.  (return key and escape will also close the menu though)

Reply:
figured this out.  actually just needed to set the QDialog to _not_ be modal!
also manually added a close event on the return/enter keys (except if the spinbox has focus... in which case we simply want return to finishEditing and blur the spinbox but not yet close the window)